### PR TITLE
Improve sampling of neutron start and scatter points in MonteCarloAbsorption

### LINKED ISSUE #30363: Fix couple of issues with bounding box determination in MonteCarloAbsorption
There are two issues with the way the MonteCarloAbsorption algorithm calculates bounding boxes that came out of some recent work on the PaalmanPingsMonteCarloAbsorption algorithm. These both relate to code in the MCInteractionVolume.cpp unit:

1) getBoundingBox() determines the area in which the neutron start point occurs. This takes the smaller of the sample bounding box and the beam profile. This has a small problem because it only considers the bounding box of the sample. In the situation where there's an environment but no sample this results in a bounding box of size zero and all neutrons start from the same point (x=y=0 I think)
2) If a workspace has a sample and an environment and you run a calculation with SimulateScatteringPointsIn=SampleOnly, the method getFullBoundingBox() sets up an active region that is the bounding box of the environment. This is unnecessarily large and I've seen some cases where the sample is much smaller than the environment where the code fails to find a point in the sample (the 5000 points limit on the number of attempts to generate a point is reached)

I think it's perhaps the case that the two methods can be replaced by a single getBoundingBox method that fixes both problems

### PR DESCRIPTION
**Pull request by DannyHindson**
**Description of work.**

Fix these two problems in the MonteCarloAbsorption algorithm (which mostly occur when its called from the workflow algorithm PaalmanPingsMonteCarloAbsorption):

- In the situation where there's an environment shape but no sample shape, the start points were all at (x,y)=(0,0) because the
start point sampling was always driven by the sample shape only

- In a situation where there's a sample and environment shape and the scattering is restricted to sampling only, the scatter point sampling was inefficient and involved a lot of throw away points because the algorithm was still using the total bounding box including the env

The system test ref files for PaalmanPingsMonteCarloAbsorption have changed because these changes alter the random number sequences used to generate the scatter points for the assc part of the output. I don't think the calculations were converged enough for the tolerance to still pass the tests so I have increased the number of scenarios on the 3 tests in this suite. This has meant all 4 workspaces in the output workspace group have changed (ass, assc, acsc, acc)

**To test:**

Run this script which runs an absorption correction calculation on a set up with a small sample and a much bigger environment. Prior to these changes, the script fails because it can't generate a scatter point in the sample in 5000 attempts because the area it's sampling in includes the environment even though "sample only" is specified:

```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

testws = CreateWorkspace([1.0,2.0,3.0],[4.0,4.0,4.0],UnitX="Wavelength")
LoadInstrument(Workspace=testws,RewriteSpectraMap=True,InstrumentName="POLARIS")
SetSample(testws,
        Geometry={'Shape': 'FlatPlate', 'Height': 0.1,
                  'Width': 0.2, 'Thick': 0.1,
                  'Center': [0.,0.,0.]},
        ContainerGeometry={'Shape': 'FlatPlateHolder', 'Height': 4.0,
                  'Width': 2.0, 'Thick': 1.0, 'FrontThick': 0.3, 'BackThick': 0.4,
                  'Center': [0.,0.,0.]},
        ContainerMaterial={'ChemicalFormula': 'Al',
                  'NumberDensity': 0.01})
SetBeam(InputWorkspace=testws,Geometry={'Shape': 'Slit', 'Width': 3.0, 'Height': 6.75})
abs = MonteCarloAbsorption(InputWorkspace=testws,EventsPerPoint=100,SimulateScatteringPointIn="SampleOnly")
```

Fixes #30363.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp [hunk 1]
```diff
@@ -61,21 +61,13 @@ MCInteractionVolume::MCInteractionVolume(
...
-/**
- * Returns the axis-aligned bounding box for the volume
- * @return A reference to the bounding box
- */
-const Geometry::BoundingBox &MCInteractionVolume::getBoundingBox() const {
-  return m_sample->getBoundingBox();
-}
-
 /**
  * Returns the axis-aligned bounding box for the volume including env
```
Comment (thomashampson) [inline Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp:65]: Comment in function header is not quite correct any more.

**Review by thomashampson — CHANGES_REQUESTED** [review]
I can confirm that the test script demonstrates that this change works as intended. Only one small suggestion regarding the comment above the `getFullBoundingBox()` function.

**Review by thomashampson — APPROVED** [review]

File: Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp (hunk 1 above)
Comment (thomashampson) [inline Framework/Algorithms/src/SampleCorrections/MCInteractionVolume.cpp:65]: Comment in function header is not quite right any more.

**Comment by martyngigg** [comment]
Merging to avoid mass-formattting conflicts.

**Pull request merged**